jwql/website/apps/jwql/monitor_pages/monitor_readnoise_bokeh.py

```diff
@@ -59,102 +72,128 @@ def load_data(self):
-                if min_val == max_val:
-                    self.refs['mean_readnoise_yr_amp{}'.format(amp)].start = min_val - 1
-                    self.refs['mean_readnoise_yr_amp{}'.format(amp)].end = max_val + 1
-                else:
-                    offset = (max_val - min_val) * .1
-                    self.refs['mean_readnoise_yr_amp{}'.format(amp)].start = min_val - offset
-                    self.refs['mean_readnoise_yr_amp{}'.format(amp)].end = max_val + offset
-
-    def update_readnoise_diff_plots(self):
-        """Updates the readnoise difference image and histogram"""
+
+            amp_plot = figure(title='Amp {}'.format(amp), width=280, height=280, x_axis_type='datetime')
+            amp_plot.xaxis[0].ticker.desired_num_ticks = 4
+
+            readnoise_vals = np.array([getattr(result, 'amp{}_mean'.format(amp)) for result in self.db.query_results])
+
+            filenames = [os.path.basename(result.uncal_filename).replace('_uncal.fits', '') for result in self.db.query_results]
+            expstarts_iso = np.array([result.expstart for result in self.db.query_results])
+            expstarts = np.array([datetime.strptime(date, '%Y-%m-%dT%H:%M:%S.%f') for date in expstarts_iso])
+            nints = [result.nints for result in self.db.query_results]
+            ngroups = [result.ngroups for result in self.db.query_results]
+
+            source = ColumnDataSource(data=dict(
+                        file=filenames,
+                        expstarts=expstarts,
+                        nints=nints,
+                        ngroups=ngroups,
+                        readnoise=readnoise_vals,
+                        ))
+
+            amp_plot.add_tools(HoverTool(tooltips=[("file", "@file"),
+                                                   ("time", "@expstarts"),
+                                                   ("nints", "@nints"),
+                                                   ("ngroups", "@ngroups"),
+                                                   ("readnoise", "@readnoise")]))
+
+            amp_plot.circle(x='expstarts', y='readnoise', source=source)
+
+            amp_plot.xaxis.axis_label = 'Date'
+            amp_plot.yaxis.axis_label = 'Mean Readnoise [DN]'
+
+            self.amp_plots.append(amp_plot)
+
+    def plot_readnoise_difference_image(self):
+        """Updates the readnoise difference image"""
 
         # Update the readnoise difference image and histogram, if data exists
-        if len(self.query_results) != 0:
-            # Get the most recent data; the entries were sorted by time when
-            # loading the database, so the last entry will always be the most recent.
-            diff_image_png = self.query_results[-1].readnoise_diff_image
+
+        self.diff_image_plot = figure(title='Readnoise Difference (most recent dark - pipeline reffile)',
+                                      height=500, width=500, sizing_mode='scale_width')
+
+        if len(self.db.query_results) != 0:
```

Comment:
What happens if len(self.db.query_results) == 0 ?

Reply:
I went ahead and moved all of the plot formatting outside of the data selection boolean here. So the plot will still show up and be formatted with no data if `if len(self.db.query_results) == 0`

Reply:
Looks good!